src/main/java/tutorlink/command/CalculateStudentGPACommand.java

```diff
@@ -0,0 +1,34 @@
+package tutorlink.command;
+
+import tutorlink.appstate.AppState;
+import tutorlink.commons.Commons;
+import tutorlink.exceptions.IllegalValueException;
+import tutorlink.result.CommandResult;
+import tutorlink.exceptions.TutorLinkException;
+
+import java.util.HashMap;
+
+public class CalculateStudentGPACommand extends Command {
```

Comment:
I am wondering if we can make calculate gpa an automatic function that is run every time a new grade is added to a student instead of having to call it via the parser. Is there a reason you considered this approach?